core/modules/layout/layout.admin.inc

```diff
@@ -2621,12 +2622,13 @@ function layout_settings_page($form, &$form_state) {
   $form['#attached']['css'][] = backdrop_get_path('module', 'layout') . '/css/layout.admin.css';
   $form['#attributes']['class'] = array('layout-settings-page');
 
-  // Get the list of layout template options.
-  $all_layout_info = layout_get_layout_template_info();
+  // Get the list of layout template options. The list needs to be rebuilt (see
+  // https://github.com/backdrop/backdrop-issues/issues/984)
+  $all_template_info = layout_get_layout_template_info(NULL, $rebuild = TRUE);
   $options = array();
   $default = array();
   $form_state['hidden_templates'] = array();
-  foreach ($all_layout_info as $template_name => $template_info) {
+  foreach ($all_template_info as $template_name => $template_info) {
```

Comment:
I see what you are saying, but this is inside a 40-line function and I've checked. Fixing this will make it easier for novice coders to understand the code. If you insist, I'm happy to revise.